Mainboard_ESP32/components/lora/lora.h

```diff
@@ -0,0 +1,266 @@
+ */
+#define PA_BOOST 0x80
+
+/*!
+  \brief Lora functions return values enum
+*/
+typedef enum {
+  LORA_OK = 0,
+  LORA_INIT_ERR,
+  LORA_WRITE_ERR,
+  LORA_TRANSMIT_ERR,
+  LORA_RECEIVE_ERR
+} lora_err_t;
+
+typedef enum {
+  LORA_GPIO_MODE_DISABLE = 0,
+  LORA_GPIO_MODE_INPUT,
+  LORA_GPIO_MODE_OUTPUT
+} lora_gpio_mode_t;
+
+typedef bool (*lora_SPI_transmit)(uint8_t _in[2], uint8_t _val[2]);
+typedef void (*lora_delay)(size_t _ms);
+typedef bool (*lora_GPIO_set_level)(uint8_t _gpio_num, uint32_t _level);
+typedef void (*lora_GPIO_pad_select_gpio)(uint8_t _gpio_num);
+typedef bool (*lora_GPIO_set_direction)(uint8_t _gpio_num,
+                                        lora_gpio_mode_t _direction);
+typedef void (*lora_log)(const char *info);
+
+typedef struct {
+  lora_SPI_transmit spi_transmit;
+  lora_delay delay;
+  lora_GPIO_set_level gpio_set_level;
+  lora_GPIO_pad_select_gpio gpio_pad_select;
+  lora_GPIO_set_direction gpio_set_direction;
+  lora_log log;
+  uint8_t rst_gpio_num;
+  uint8_t cs_gpio_num;
+  uint8_t d0_gpio_num;
+  int16_t implicit;
+  int32_t frequency;
+} lora_struct_t;
+
+/*!
+ * \brief Perform hardware initialization.
+ */
+lora_err_t lora_init(lora_struct_t *lora);
+
+/*!
+ * \brief Create default config and set parameters
+ */
+lora_err_t lora_default_config(lora_struct_t *lora);
+
+/*!
+ * \brief Write a value to a register.
+ * \param reg Register index.
+ * \param val Value to write.
+ * \return lora_err_t value
+ */
+lora_err_t lora_write_reg(lora_struct_t *lora, int16_t reg, int16_t val);
+
+/*!
+ * \brief Read the current value of a register.
+ * \param reg Register index.
+ * \returns Value of the register.
+ */
+uint8_t lora_read_reg(lora_struct_t *lora, int16_t reg);
+
+/*!
+ * \brief Perform physical reset on the Lora chip
+ */
+void lora_reset(lora_struct_t *lora);
+
+/*!
+ * \brief Configure explicit header mode.
+ * Packet size will be included in the frame.
+ */
+void lora_explicit_header_mode(lora_struct_t *lora);
+
+/*!
+ * \brief Configure implicit header mode.
+ * All packets will have a predefined size.
+ * \param size Size of the packets.
+ */
+void lora_implicit_header_mode(lora_struct_t *lora, int16_t size);
+
+/*!
+ * \brief Sets the radio transceiver in idle mode.
+ * \note Must be used to change registers and access the FIFO.
+ */
+void lora_idle(lora_struct_t *lora);
+
+/*!
+ * \brief Sets the radio transceiver in sleep mode.
+ * \note Low power consumption and FIFO is lost.
+ */
+void lora_sleep(lora_struct_t *lora);
+
+/*!
+ * \brief Sets the radio transceiver in receive mode.
+ * \note Incoming packets will be received.
+ */
+void lora_set_receive_mode(lora_struct_t *lora);
+
+/*!
+ * \brief Configure power level for transmission
+ * \param level 2-17, from least to most power
+ */
+void lora_set_tx_power(lora_struct_t *lora, int16_t level);
+
+/*!
+ * \brief Set carrier frequency.
+ * \param frequency Frequency in Hz
+ */
+void lora_set_frequency(lora_struct_t *lora, int32_t frequency);
```

Comment:
jaka jednostka?

Reply:
\param frequency Frequency in Hz jest w dokumentacji